docs/quest-definition/schema.json

```
@@ -182,76 +166,52 @@
         "quest_answer_validation": {
           "description": "Validation for the answer. Only applicable for Numeric type questions.",
           "type": "object",
+          "additionalProperties": false,
           "properties": {
             "min": {
               "description": "Minimum value for the answer",
               "type": "number",
-              "examples": [
-                0,
-                1,
-                2
-              ]
+              "examples": [0, 1, 2]
             },
             "max": {
               "description": "Maximum value for the answer",
               "type": "number",
-              "examples": [
-                0,
-                1,
-                2
-              ]
+              "examples": [0, 1, 2]
             }
           }
         },
         "quest_answer_dependency": {
-          "description": "",
-          "type": "object",
-          "properties": {
-            "question_id": {
-              "description": "The Quest ID of the question that this questions visibility is dependent on.",
-              "type": "number",
-              "examples": [
-                1,
-                2,
-                3
-              ]
+          "description": "Defines the dependency of this question on another question. If the required_value of the dependent question is selected, then this question will be visible. Otherwise, it will be hidden. It can be a single object or an array of objects to define multiple dependencies (AND logic).",
+          "oneOf": [
+            {
+              "$ref": "#/definitions/dependency"
             },
-            "required_value": {
-              "description": "The value of the dependent question that will make this question visible.",
-              "oneOf": [
-                {
-                  "type": "string"
-                },
-                {
-                  "type": "array",
-                  "items": {
-                    "type": "string"
-                  }
-                }
-              ],
-              "examples": [
-                "yes",
-                "0",
-                [
-                  "1",
-                  "2"
-                ],
-                [
-                  "yes",
-                  "no"
-                ]
-              ]
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/dependency"
+              },
+              "minItems": 1
             }
-          }
+          ]
         }
       },
       "required": [
         "quest_id",
         "quest_title",
         "quest_description",
         "quest_type",
-        "quest_tag",
-        "quest_answer_choices"
+        "quest_tag"
+      ],
+      "allOf": [
+        {
+          "if": {
+            "properties": { "quest_type": { "const": "TextEntry" } }
+          },
+          "else": {
```

Plugging in @uw-amy-bordenave's [comment](https://github.com/TaskarCenterAtUW/tdei-tools/pull/25#issuecomment-3309953350).

```suggestion
          "then": {
```